extending/sanitizers.rst

```diff
@@ -0,0 +1,170 @@
+passed to the compiler as flags and, depending on if you are using
+`clang <https://clang.llvm.org/docs/UsersManual.html#controlling-code-generation>`_ or
+`gcc <https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html>`_, different sanitizers are
+supported.
+
+Here we explain different options on how to model and use sanitizers with your Conan packages.
+
+Adding custom settings
+----------------------
+
+If you want to model the sanitizer options so that the package id is affected by them, you have to
+introduce new settings in the *settings.yml* file (see :ref:`custom_settings` section for more
+information).
+
+Sanitizer options should be modeled as sub-settings of the compiler. Depending on how you want to
+combine the sanitizers you have two choices.
+
+Adding a list of commonly used options
+######################################
+
+If you have a fixed set of sanitizers or combinations of them that are the ones you usually set for
+your builds you can add the sanitizers as a list of options. An example for *apple-clang* would be
+like this:
+
+.. code-block:: yaml
+    :caption: *settings.yml*
+    :emphasize-lines: 6
+    
+    apple-clang:
+        version: ["5.0", "5.1", "6.0", "6.1", "7.0", "7.3", "8.0", "8.1", 
+                  "9.0", "9.1", "10.0", "11.0"]
+        libcxx: [libstdc++, libc++]
+        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
+        sanitizer: [None, Address, Thread, Memory, UndefinedBehavior, AddressUndefinedBehavior]
+
+Here you have modeled the use of ``-fsanitize=address``, ``-fsanitize=thread``,
+``-fsanitize=memory``, ``-fsanitize=undefined`` and the combination of ``-fsanitize=address`` and
+``-fsanitize=undefined``. Note that for example, for clang it is not possible to combine more than
+one of the ``-fsanitize=address``, ``-fsanitize=thread``, and ``-fsanitize=memory`` checkers in the
+same program.
+
+Adding thread sanitizer for a :command:`conan install`, in this case, could be done by calling
+:command:`conan install .. -s compiler.sanitizer=Thread`
+
+Adding different options to combine
+###################################
+
+Another option would be to add the sanitizer options as multiple ``True`` or ``None`` fields so that
+they can be freely combined later. An example of that for the previous sanitizer options would be as
+follows:
+
+.. code-block:: yaml
+    :caption: *settings.yml*
+    :emphasize-lines: 6,7,8
+
+    apple-clang:
+        version: ["5.0", "5.1", "6.0", "6.1", "7.0", "7.3", "8.0", 
+                  "8.1", "9.0", "9.1", "10.0", "11.0"]
+        libcxx: [libstdc++, libc++]
+        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
+        address_sanitizer: [None, True]
+        thread_sanitizer: [None, True]
+        undefined_sanitizer: [None, True]
+
+Then, you can add different sanitizers calling, for example, to :command:`conan install ..
+-s compiler.address_sanitizer=True -s compiler.undefined_sanitizer=True`
```

Comment:
A drawback of this approach is that not all the combinations will be valid or will make sense, but it is up to the consumer to use it correctly.

Reply:
Thank you, I think that's worth mentioning